backend/wmg/pipeline/expression_summary_default.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import os
+
+import pandas as pd
+import tiledb
+
+from backend.wmg.data.schemas.cube_schema_default import (
+    expression_summary_indexed_dims,
+    expression_summary_non_indexed_dims,
+    expression_summary_schema,
+)
+from backend.wmg.data.snapshot import (
+    EXPRESSION_SUMMARY_CUBE_NAME,
+    EXPRESSION_SUMMARY_DEFAULT_CUBE_NAME,
+)
+from backend.wmg.data.tiledb import create_ctx
+from backend.wmg.data.utils import log_func_runtime
+from backend.wmg.pipeline.constants import (
+    EXPRESSION_SUMMARY_AND_CELL_COUNTS_CUBE_CREATED_FLAG,
+    EXPRESSION_SUMMARY_DEFAULT_CUBE_CREATED_FLAG,
+)
+from backend.wmg.pipeline.errors import PipelineStepMissing
+from backend.wmg.pipeline.utils import (
+    create_empty_cube_if_needed,
+    load_pipeline_state,
+    write_pipeline_state,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@log_func_runtime
+def create_expression_summary_default_cube(*, corpus_path: str):
+    """
+    Create the default expression summary cube. The default expression summary cube is an aggregation across
+    non-default dimensions in the expression summary cube.
+    """
+    pipeline_state = load_pipeline_state(corpus_path=corpus_path)
+
+    if not pipeline_state.get(EXPRESSION_SUMMARY_AND_CELL_COUNTS_CUBE_CREATED_FLAG):
+        raise PipelineStepMissing("expression_summary")
+
+    logger.info("Creating the default expression summary cube.")
+    expression_summary_uri = os.path.join(corpus_path, EXPRESSION_SUMMARY_CUBE_NAME)
+    expression_summary_default_uri = os.path.join(corpus_path, EXPRESSION_SUMMARY_DEFAULT_CUBE_NAME)
+
+    ctx = create_ctx()
+    with tiledb.scope_ctx(ctx):
+        dfs = []
+        with tiledb.open(expression_summary_uri, "r") as cube:
+            for row in cube.query(return_incomplete=True).df[:]:
+                dfs.append(row)
+        expression_summary_df = pd.concat(dfs, axis=0)
+
+        expression_summary_df_default = (
+            expression_summary_df.groupby(expression_summary_indexed_dims + expression_summary_non_indexed_dims)
```

Comment:
PyArrow tables also can perform `groupby`. I'm curious if it would be faster to stay fully in the pyarrow "space". Just a thought, and hardly important if this working efficiently enough.

Reply:
Left a TODO! I think the runtime is fast enough, would be interesting to see if Pyarrow groupby incurs less memory overhead, though.